lib/Doctrine/DBAL/Schema/Identifier.php

```diff
@@ -35,9 +35,14 @@ class Identifier extends AbstractAsset
      * Constructor.
      *
      * @param string $identifier Identifier name to wrap.
+     * @param bool   $quote      Whether to force quoting the given identifier.
      */
-    public function __construct($identifier)
+    public function __construct($identifier, $quote = false)
     {
         $this->_setName($identifier);
+
+        if ($quote && ! $this->_quoted) {
+            $this->_setName('"' . $this->getName() . '"');
```

Comment:
The problem is always that if a name existing in the db contains an unmatched quote character in it, the item essentially becomes untouchable and breaks DBAL's schema system completely. I don't care that much if people have broken crazy table names in their databases, but the way AbstractAsset::trimQuotes works is incorrect in an academic sense.

I think in dbal 3 we should probably just quote everything and call it a day.